packages/preview-middleware-client/src/adp/utils.ts

```diff
@@ -26,3 +35,28 @@ export function createDeferred<T>(): Deferred<T> {
 
     return { promise, resolve, reject };
 }
+
+/**
+ * Checks if the fragment name associated with a command matches the specified fragment name.
+ *
+ * @param {FlexCommand} command - The command object containing the prepared change to be examined.
+ * @param {string} fragmentName - The name of the fragment to match against the command's change.
+ * @returns {boolean} Returns true if the command's change contains a fragment path that matches
+ *                    the specified fragment name; otherwise, returns false.
+ */
+export function matchesFragmentName(command: FlexCommand, fragmentName: string): boolean {
+    const change = command.getPreparedChange().getDefinition() as unknown as FragmentChange;
+    return change.content?.fragmentPath?.includes(`${fragmentName}.fragment.xml`) || false;
```

Comment:
Can we cast direct to FragmentChange?

Reply:
No, we cannot because getDefinition API returns a different type which is more generic for all of the different changes. But from this one I only need content.fragmentPath which obviously does not exist on other changes. The correct way here is to cast to unknown first and then to whatever we want the interface to be.